### ISSUE TITLE
Motioneye freezes with "mjpg client timed out" every few days

### ISSUE THREAD

**Issue by Dinth** [ccrisan/motioneye#1544]
Im running Motioneye official docker container.
Normally it works perfectly fine with 4-5 network cameras but every few days Motioneye all streams break down. Some cameras are showing image frozen several hours ago and some streams are completely offline.
Logs show
```
   ERROR: mjpg client timed out receiving data for camera 3 on port 8083
   ERROR: mjpg client timed out receiving data for camera 1 on port 8081
   ERROR: mjpg client timed out receiving data for camera 1 on port 8081
   ERROR: mjpg client timed out receiving data for camera 1 on port 8081
   ERROR: mjpg client timed out receiving data for camera 1 on port 8081
   ERROR: mjpg client timed out receiving data for camera 1 on port 8081
```

**Comment by wzbart** [comment]
> Im running Motioneye official docker container.
> Normally it works perfectly fine with 4-5 network cameras but every few days Motioneye all streams break down. Some cameras are showing image frozen several hours ago and some streams are completely offline.
> Logs show
> 
> ```
>    ERROR: mjpg client timed out receiving data for camera 3 on port 8083
>    ERROR: mjpg client timed out receiving data for camera 1 on port 8081
>    ERROR: mjpg client timed out receiving data for camera 1 on port 8081
>    ERROR: mjpg client timed out receiving data for camera 1 on port 8081
>    ERROR: mjpg client timed out receiving data for camera 1 on port 8081
>    ERROR: mjpg client timed out receiving data for camera 1 on port 8081
> ```

Have you found the solution? I'm running in to very same issue.

**Comment by Dinth** [comment]
No i havent so far. I have removed a couple of cameras from motion eye leaving just three. The situation has improved slightly.
It would be nice to have some kind of watchdog build into motioneye which checks if ffmpeg processes are working.

**Comment by Legolaszstudio** [comment]
It's funny how it hasn't been fixed **yet**.....

**Comment by MichielvanBeers** [comment]
Agreed with @Dinth. Is there anyway how we can for example make our own script that checks the motioneye log for this error? Since motioneye is aware that the issue takes place, we should be able to restart the service, right?

**Comment by STUKguy** [comment]
Did anyone find answer for this?

**Comment by starbasessd** [comment]
Are you getting it? I have run docker on a VM, but never had it crash. What version of motion & motionEye & OS is showing in the docker motionEye WebGUI?

**Comment by MichielvanBeers** [comment]
Not sure if that question was directed at me, but I currently have it running stable by triggering a cronjob every night that reboots the device. I run the latest version: 

motionEye Version | 0.42.1
Motion Version | 4.2.2
OS Version | motionEyeOS 20200606

**Comment by Woodpeckercz** [comment]
> Not sure if that question was directed at me, but I currently have it running stable by triggering a cronjob every night that reboots the device. I run the latest version:
> 
> motionEye Version | 0.42.1
> Motion Version | 4.2.2
> OS Version | motionEyeOS 20200606

I don't know how to create cron jobs yet, would you mind sharing your code for restarting the camera every night ?
Seems like my camera looses connection every 24-48h.

**Comment by MichielvanBeers** [comment]
Yeah no problem:

Make sure that SSH is turned on in the Service settings. Login via SSH on your motioneye server with root and the password that you've set for the admin user in the webinterface. 

After logged in, type `crontab -e` and paste the following line in the crontab page: `0 4 * * * reboot now >/dev/null 2>&1` . Press `ctrl + X` and save the changes. Your device should now reboot everyday at 4:00 am if all is well.

**Comment by Woodpeckercz** [comment]
> Yeah no problem:
> 
> Make sure that SSH is turned on in the Service settings. Login via SSH on your motioneye server with root and the password that you've set for the admin user in the webinterface.
> 
> After logged in, type `crontab -e` and paste the following line in the crontab page: `0 4 * * * reboot now >/dev/null 2>&1` . Press `ctrl + X` and save the changes. Your device should now reboot everyday at 4:00 am if all is well.

Thank you so much.

**Comment by leprechaun17** [comment]
> crontab -e

How do you perform this when motioneye is installed as a docker container on unraid?

**Comment by MichielvanBeers** [comment]
@leprechaun17 you should run the command inside of the Docker container. I don't use Unraid but the command line and then it would go like [this](https://www.digitalocean.com/community/tutorials/how-to-use-docker-exec-to-run-commands-in-a-docker-container). Keep in mind that Docker containers are stateless and that restarting it, will remove the changes you've made. You can find some Unraid specific info [here](https://forums.unraid.net/topic/133840-docker-commands-inside-a-container/).
